internal/controller/clusterdeployment_controller.go

```diff
@@ -505,48 +505,65 @@ func getProjectPolicyRefs(mc *kcm.ClusterDeployment, cred *kcm.Credential) []sve
 	}
 }
 
+func (*ClusterDeploymentReconciler) initServicesConditions(cd *kcm.ClusterDeployment) (changed bool) {
+	for _, typ := range [3]string{kcm.SveltosProfileReadyCondition, kcm.FetchServicesStatusSuccessCondition, kcm.ServicesReferencesValidationCondition} {
+		if apimeta.SetStatusCondition(&cd.Status.Conditions, metav1.Condition{
+			Type:               typ,
+			Status:             metav1.ConditionUnknown,
+			Reason:             kcm.ProgressingReason,
+			ObservedGeneration: cd.Generation,
+		}) {
+			changed = true
+		}
+	}
+
+	return changed
+}
+
+func (*ClusterDeploymentReconciler) setCondition(cd *kcm.ClusterDeployment, typ string, err error) (changed bool) { //nolint:unparam // readability
+	reason, cstatus, msg := kcm.SucceededReason, metav1.ConditionTrue, ""
+	if err != nil {
+		reason, cstatus, msg = kcm.FailedReason, metav1.ConditionFalse, err.Error()
+	}
+
+	return apimeta.SetStatusCondition(&cd.Status.Conditions, metav1.Condition{
+		Type:    typ,
+		Status:  cstatus,
+		Reason:  reason,
+		Message: msg,
+	})
+}
+
 // updateServices reconciles services provided in ClusterDeployment.Spec.ServiceSpec.
 func (r *ClusterDeploymentReconciler) updateServices(ctx context.Context, cd *kcm.ClusterDeployment) (_ ctrl.Result, err error) {
 	l := ctrl.LoggerFrom(ctx)
 	l.Info("Reconciling Services")
 
+	r.initServicesConditions(cd)
+
+	{
```

Comment:
Why these extra curly braces?

Reply:
To declare the vars in this particular block of the code without affecting the rest of the method(s)